### PR TITLE
MOTOR-897 Set up flake8

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Review by juliusgeo — APPROVED** [review]
LGTM!

File: motor/core.py [hunk 1]
```diff
@@ -34,7 +33,7 @@
 from pymongo.driver_info import DriverInfo
 from pymongo.encryption import ClientEncryption
 
-from motor.docstrings import *
+import motor.docstrings as docstrings
```
Comment (ShaneHarvey) [inline motor/core.py:36]: `from motor import docstrings`?

File: synchro/__init__.py [hunk 2]
```diff
@@ -298,21 +298,15 @@ def __new__(mcls, name, _bases, attrs):
     return type.__new__(_metaclass, str("dummy"), (), {})
 
 
-class Synchro(with_metaclass(SynchroMeta)):
```
Comment (ShaneHarvey) [inline synchro/__init__.py:301]: Can we also remove with_metaclass now?

File: synchro/__init__.py [hunk 3]
```diff
@@ -424,7 +418,7 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.synchronize(self.delegate.end_session)
 
     def with_transaction(self, *args, **kwargs):
-        raise unittest.SkipTest("MOTOR-606 Synchro does not support " "with_transaction")
+        raise unittest.SkipTest("MOTOR-606 Synchro does not supportwith_transaction")
```
Comment (ShaneHarvey) [inline synchro/__init__.py:421]: missing space.

File: synchro/__init__.py [hunk 4]
```diff
@@ -618,7 +612,7 @@ def __init__(self, delegate):
         super().__init__(delegate)
 
     def next(self):
-        motor_grid_out = super().next()
+        motor_grid_out = next(super())
```
Comment (ShaneHarvey) [inline synchro/__init__.py:615]: Why not `super().next()`?

File: test/tornado_tests/test_motor_auth.py [hunk 5]
```diff
@@ -70,7 +70,7 @@ async def test_scram(self):
                     await client.scramtestdb.collection.insert_one({})
 
         # No mechanism specified, always works.
-        for user, mechanism, should_work in [
+        for user, _mechanism, _should_work in [
```
Comment (ShaneHarvey) [inline test/tornado_tests/test_motor_auth.py:73]: Is this a bug? _mechanism and _should_work appear unused

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: motor/core.py (hunk 1 above)
Comment (blink1073) [reply motor/core.py:36]: Haha yeah, will do

File: synchro/__init__.py (hunk 2 above)
Comment (blink1073) [reply synchro/__init__.py:301]: Done

File: synchro/__init__.py (hunk 3 above)
Comment (blink1073) [reply synchro/__init__.py:421]: fixed

File: synchro/__init__.py (hunk 4 above)
Comment (blink1073) [reply synchro/__init__.py:615]: `B305 `.next()` is not a thing on Python 3. Use the `next()` builtin. For Python 2 compatibility, use `six.next()`.`

File: test/tornado_tests/test_motor_auth.py (hunk 5 above)
Comment (blink1073) [reply test/tornado_tests/test_motor_auth.py:73]: adding an underscore indicates to flake that they are not used, I'll just make them regular underscores.

Comment (ShaneHarvey) [reply test/tornado_tests/test_motor_auth.py]: Yeah I get that but I'm wondering why they are unused. Can we just simplify this to:
```
for user in ("sha1", "sha256", "both"):...
```

Comment (blink1073) [reply test/tornado_tests/test_motor_auth.py]: Ah, right, done

File: synchro/__init__.py (hunk 4 above)
Comment (ShaneHarvey) [reply synchro/__init__.py:605]: But Cursor.next() it is a thing in pymongo. That's what's confusing to me. I think we should disable this check as long as we support next() methods.

Comment (blink1073) [reply synchro/__init__.py:605]: Yeah I'll disable `B305` globally here as well.

File: synchro/__init__.py (hunk 2 above)
Comment (ShaneHarvey) [reply synchro/__init__.py:301]: These changes cause this error in the synchro tests:
```
 [2022/02/17 01:40:50.485] Traceback (most recent call last):
 [2022/02/17 01:40:50.485]   File "/opt/python/3.7/lib/python3.7/runpy.py", line 183, in _run_module_as_main
 [2022/02/17 01:40:50.485]     mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
 [2022/02/17 01:40:50.485]   File "/opt/python/3.7/lib/python3.7/runpy.py", line 109, in _get_module_details
 [2022/02/17 01:40:50.485]     __import__(pkg_name)
 [2022/02/17 01:40:50.485]   File "/data/mci/25c39eae21d0be14b9a9ad83b5be38eb/src/synchro/__init__.py", line 291, in <module>
 [2022/02/17 01:40:50.485]     class Synchro(meta=SynchroMeta):
 [2022/02/17 01:40:50.485] TypeError: __init_subclass__() takes no keyword arguments
```

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: synchro/__init__.py (hunk 2 above)
Comment (blink1073) [reply synchro/__init__.py:301]: Fixed, used the wrong argument name

**Comment by ShaneHarvey** [comment]
Waiting for the synchro task to complete.

**Comment by blink1073** [comment]
Looks like we're out of synch on parameters: `TypeError: watch() got an unexpected keyword argument 'comment`

**Comment by ShaneHarvey** [comment]
Yes that's expected. I just wanted to make sure that the synchro tests fail in the same way as before.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**